File: src/main/java/dev/paigewatson/layoutmaster/models/layout/IndustryTrack.java
package dev.paigewatson.layoutmaster.models.layout;

import dev.paigewatson.layoutmaster.models.rollingstock.FreightCar;

import java.util.ArrayList;

public class IndustryTrack
{
    private final String trackName;
    private final int maximumNumberOfCars;
    private final ArrayList<FreightCar> carsAtIndustry;

    public IndustryTrack(String trackName, int maximumNumberOfCars)
    {
        this.trackName = trackName;
        this.maximumNumberOfCars = maximumNumberOfCars;
        carsAtIndustry = new ArrayList<>();
    }

    public String name()
    {
        return trackName;
    }

    public boolean hasOpenSlots()
    {
        return carsAtIndustry.size() < maximumNumberOfCars;
    }

    public void setOutCar(FreightCar freightCar)
    {
        carsAtIndustry.add(freightCar);
    }
}

File: src/main/java/dev/paigewatson/layoutmaster/models/layout/IndustryTrack.kt
package dev.paigewatson.layoutmaster.models.layout

import dev.paigewatson.layoutmaster.models.rollingstock.FreightCar
import java.util.*

class IndustryTrack(private val trackName: String, private val maximumNumberOfCars: Int) {
    private val carsAtIndustry = ArrayList<FreightCar>()
    fun name(): String {
        return trackName
    }

    fun hasOpenSlots(): Boolean {
        return carsAtIndustry.size < maximumNumberOfCars
    }

    fun setOutCar(freightCar: FreightCar) {
        carsAtIndustry.add(freightCar)
    }

    override fun toString(): String {
        return "IndustryTrack{" +
                "trackName='" + trackName + '\'' +
                ", maximumNumberOfCars=" + maximumNumberOfCars +
                ", carsAtIndustry=" + carsAtIndustry +
                '}'
    }
}
